packages/@react-stately/color/src/useColorAreaState.ts

```diff
@@ -194,16 +204,16 @@ export function useColorAreaState(props: ColorAreaProps): ColorAreaState {
       return {x, y};
     },
     incrementX(stepSize) {
-      setXValue(snapValueToStep(xValue + stepSize, minValueX, maxValueX, stepSize));
+      setXValue(xValue + stepSize > maxValueX ? maxValueX : snapValueToStep(xValue + stepSize, minValueX, maxValueX, xChannelStep));
     },
     incrementY(stepSize) {
-      setYValue(snapValueToStep(yValue + stepSize, minValueY, maxValueY, stepSize));
+      setYValue(yValue + stepSize > maxValueY ? maxValueY : snapValueToStep(yValue + stepSize, minValueY, maxValueY, yChannelStep));
     },
     decrementX(stepSize) {
-      setXValue(snapValueToStep(xValue - stepSize, minValueX, maxValueX, stepSize));
+      setXValue(snapValueToStep(xValue - stepSize, minValueX, maxValueX, xChannelStep));
     },
     decrementY(stepSize) {
-      setYValue(snapValueToStep(yValue - stepSize, minValueY, maxValueY, stepSize));
+      setYValue(snapValueToStep(yValue - stepSize, minValueY, maxValueY, yChannelStep));
     },
```

Comment:
I still think increment/decrement should snap to `stepSize` rather than `xChannelStep` or `yChannelStep`. So that snapping is based on the interval by which one increments or decrements.

Reply:
For example step=5 pageStep=8

step start at 0 => 5

would it be
pageStep start at 0 => 8 & start at 5 (from the regular step) => 13
or is it
pageStep start at 0 => 8 & start at 5 (from the regular step) => 16


Native sliders, when supplied a step, cannot have a value that does not fall on a multiple of step. You can see this clearly in the Chrome Date picker. I think this is more predictable.
pageStep start at 0 => 8 & start at 5 (from the regular step) => 15

Or are you suggesting that color sliders are different from regular sliders?

Reply:
Caught up in the other discussion on the same topic. Devon had an idea, we keep the page step size out of the API and just treat it as a multiple of the step. We currently don't support page step size in our Sliders, so this makes a fair amount of sense

Reply:
I can see where you're coming from if the uses a step size that is not a factor of the page size. With ColorSliders, the step is usually 1, or 0.01 for alpha, and default for pageSize will always snap to step. Can do something like this, using modulus?
```ts
    incrementX(stepSize) {
      setXValue(xValue + stepSize > maxValueX ? maxValueX : snapValueToStep(xValue + stepSize, minValueX, maxValueX, stepSize > xChannelStep && stepSize % xChannelStep === 0 ? stepSize : xChannelStep));
    },
    incrementY(stepSize) {
      setYValue(yValue + stepSize > maxValueY ? maxValueY : snapValueToStep(yValue + stepSize, minValueY, maxValueY, stepSize > yChannelStep && stepSize % yChannelStep === 0 ? stepSize : yChannelStep));
    },
    decrementX(stepSize) {
      setXValue(snapValueToStep(xValue - stepSize, minValueX, maxValueX, stepSize > xChannelStep && stepSize % xChannelStep === 0 ? stepSize : xChannelStep));
    },
    decrementY(stepSize) {
      setYValue(snapValueToStep(yValue - stepSize, minValueY, maxValueY, stepSize > yChannelStep && stepSize % yChannelStep === 0 ? stepSize : yChannelStep));
    },
```

Reply:
it's getting too complicated I think, see previous comment about doing away with it in our API
That way we can make sure it's a multiple of the step and eliminate that worry.

Reply:
I guess my question would be, what multiple of a step would make the most sense for color channels? The reason for returning `pageSize` as part of the color channel range object returned by `Color.getChannelRange` was so that we could have more useful page stepping than simply dividing the range by 10. For example for the hue channel, a pageSize of 15 seems to make sense for snapping to primary, secondary and tertiary colors around the color wheel (BTW, we should use `getChannelRange('hue').pageSize` instead of the `PAGE_MIN_STEP_SIZE` constant in `useColorWheel`). I suggested 0x11 as a pageSize for RGB channels because it divides 0xFF evenly, and increments channel values in a way that makes sense for web developers 0x11, 0x22, 0x33, 0x44, ..., 0xDD, 0xEE, 0xFF.

Reply:
Another option would be to eliminate `step` and `pageStep` from the public API for color controls, and use values that make sense for each color channel internally. Then we wouldn't have to worry about a `pageSize` that is not a multiple of a `step` because we would control it.

Reply:
Maybe this brings up if step should be configurable too?

What is the actual use case for setting step sizes in a color area or color slider? Maybe we should consider that color sliders are different than sliders

RGB 255 is not many values. Having the step size always be 1 and the page size always be 11 makes a lot of sense, you can get quickly to any value and you are always evenly and predictably within the range.

For HSL 0-360 and 0-1 are known ranges that will not change, a step of 1 and 0.01 respectively make sense and page sizes are easy to figure out from there as well.

Reply:
I still think that snapping to  `stepSize`, the `pageStep`, is preferable to snapping to `step`, the channel `step`, and now that we can be sure that the channel `step` is a factor of the `stepSize`, I don't se why we shouldn't do it this way.